src/main/java/io/numaproj/pulsar/consumer/PulsarSource.java

```
@@ -137,16 +138,47 @@ public void ack(AckRequest request) {
     @Override
     public long getPending() {
         try {
-            // If changing to support multiple topics, need to update this
+            // If changing to support multiple topics, we need to update this
             Set<String> topicNames = (Set<String>) pulsarConsumerProperties.getConsumerConfig().get("topicNames");
             String topicName = (String) topicNames.iterator().next(); // Assumes there is only one topic name in the set
             String subscriptionName = (String) pulsarConsumerProperties.getConsumerConfig().get("subscriptionName");
```

Is this a safe type conversion? meaning if user specify an integer, do we fail early?

---

https://github.com/numaproj-contrib/apache-pulsar-java/blob/e3997d0622e95ad9633e949a41fc5f4b91353233/src/main/java/io/numaproj/pulsar/config/consumer/PulsarConsumerProperties.java#L33

TopicName is a required field for the consumer. In the PostConstruct method, we always turn the value provided for the topicNames key in the configMap into a string.

---

Added type check in PulsarConsumerProperties!